File: studycollab-app/app/src/main/java/com/example/studycollab/data/model/Submission.kt
package com.example.studycollab.data.model

data class Submission(
    val _id: String,
    val assignmentId: String,
    val groupId: GroupInfo, // Populated from backend
    val fileUrl: String,
    val grade: String?,
    val feedback: String?,
    val submittedAt: String
)

data class GroupInfo(
    val _id: String,
    val groupName: String
)

data class SubmissionState(
    val groupName: String,
    val status: String,
    val grade: String?
)

File: backend/src/models/Submission.js
const mongoose = require('mongoose');

const SubmissionSchema = new mongoose.Schema({
    assignmentId: { 
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'Assignment', 
        required: true 
    },
    groupId: { 
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'StudyGroup', 
        required: true 
    },
    courseId: { 
        type: mongoose.Schema.Types.ObjectId, 
        ref: 'Course', 
        required: true 
    },
    fileUrl: { 
        type: String, 
        required: true
    },
    grade: { 
        type: String, 
        default: null 
    },
    feedback: { 
        type: String, 
        default: "" 
    },
    taskReport: [{
        studentId: { type: mongoose.Schema.Types.ObjectId, ref: 'User' },
        studentName: { type: String },
        contribution: { type: String }
    }],
    submittedAt: { 
        type: Date, 
        default: Date.now 
    }
});

module.exports = mongoose.model('Submission', SubmissionSchema);